File: core/src/main/java/dev/jason/project/spring/vc_server/core/model/Message.java
package dev.jason.project.spring.vc_server.core.model;

import java.util.Map;

public record Message(String from, String to, String text, String timestamp) {

    public Map<String, String> asMap() {
        return Map.of(
            "received_from", from,
            "to", to,
            "text", text,
            "timestamp", timestamp
        );
    }
}

File: core/src/main/java/dev/jason/app/compose/vaultchat/core/model/message/Message.kt
package dev.jason.app.compose.vaultchat.core.model.message

import java.time.LocalDateTime

data class Message(
    val id: String? = null, // only null while sending to backend, not while receiving from backend.
    val from: String,
    val fromDisplayName: String?,
    val to: String,
    val text: String,
    val timestamp: LocalDateTime
) {
    constructor(
        id: String? = null,
        from: String,
        to: String,
        text: String,
        timestamp: LocalDateTime
    ) : this(id, from, null, to, text, timestamp)
}